src/System.IO/tests/StringWriter/StringWriterTests.netcoreapp.cs

```diff
@@ -36,5 +37,61 @@ public partial class StringWriterTests
             await Assert.ThrowsAnyAsync<OperationCanceledException>(() => writer.WriteAsync(Memory<char>.Empty, new CancellationToken(true)));
             await Assert.ThrowsAnyAsync<OperationCanceledException>(() => writer.WriteLineAsync(Memory<char>.Empty, new CancellationToken(true)));
         }
+
+        [Fact]
+        public void TestWriteStringBuilder()
+        {
+            StringBuilder sb = getSb();
+            StringWriter sw = new StringWriter();
+            sw.Write(sb);
+            Assert.Equal(sb.ToString(), sw.ToString());
+        }
+
+        [Fact]
+        public async Task TestWriteAsyncStringBuilder()
+        {
+            StringBuilder sb = getSb();
+            StringWriter sw = new StringWriter();
+            await sw.WriteAsync(sb);
+            Assert.Equal(sb.ToString(), sw.ToString());
+        }
+
+        [Fact]
+        public async Task TestWriteAsyncStringBuilderCancelled()
+        {
+            StringBuilder sb = getSb();
+            StringWriter sw = new StringWriter();
+            CancellationTokenSource cts = new CancellationTokenSource();
+            cts.Cancel();
+            await Assert.ThrowsAsync<TaskCanceledException>(async () => await sw.WriteAsync(sb, cts.Token));
```

Comment:
This will pass even if the cancellation occurs asynchronously.  It'd be better to write it as something like:
```C#
Assert.Equal(TaskStatus.Canceled, sw.WriteAsync(sb, cts.Token).Status);
```

Reply:
ok i amended code...but it's not clear to me why. Can you explain?

Reply:
> it's not clear to me why. Can you explain?

The version you had would pass whether the implementation completed with the exception synchronously or whether the implementation completed with the exception asynchronously.  In this case, since the cancellation occurred prior to the method being called, we want it to always complete in this case synchronously, and so we want to verify that as best we can, which my suggested version does.